### ISSUE TITLE
samgeo.generate does not accept np.ndarray as source parameter

### ISSUE THREAD

**Issue by glemoine62** [opengeos/segment-geospatial#137]
### Environment Information

-   samgeo version: 0.8.4
-   Python version: 3.8.10
-   Operating System: ubuntu 20.04

### Description

Running samgeo.generate with a np.ndarray as source crashes. Looks like samgeo/common.py is expecting a fpath or file (only).

### What I Did

```
from samgeo import SamGeo
import sys
import cv2

data = cv2.imread(sys.argv[1] + ".tif")

sam = SamGeo(
    model_type="vit_h",
    checkpoint='sam_vit_h_4b8939.pth',
    sam_kwargs=None
)

sam.generate(data, output=f"{sys.argv[1]}_masks.tif", foreground=True, unique=True)

```

dumps:

```
Traceback (most recent call last):
  File "test2.py", line 13, in <module>
    sam.generate(data, output=f"{sys.argv[1]}_masks.tif", foreground=True, unique=True)
  File "/home/eouser/.local/lib/python3.8/site-packages/samgeo/samgeo.py", line 238, in generate
    self.save_masks(
  File "/home/eouser/.local/lib/python3.8/site-packages/samgeo/samgeo.py", line 323, in save_masks
    array_to_image(self.objects, output, self.source, **kwargs)
  File "/home/eouser/.local/lib/python3.8/site-packages/samgeo/common.py", line 1241, in array_to_image
    with rasterio.open(source) as src:
  File "/home/eouser/.local/lib/python3.8/site-packages/rasterio/env.py", line 437, in wrapper
    return f(*args, **kwds)
  File "/home/eouser/.local/lib/python3.8/site-packages/rasterio/__init__.py", line 157, in open
    raise TypeError("invalid path or file: {0!r}".format(fp))
TypeError: invalid path or file: array([[[ 50,  59,  61],
        [ 39,  49,  49],...
```

**Comment by giswqs** [comment]
Thank you for reporting. It has been fixed in #139. The `generate` function already supports `np.ndarray` input. The error was caused when trying to save the result as a GeoTIFF. Since no source image is provide, the resulting image will not contain crs.

**Comment by giswqs** [comment]
Please run the following code and restart kernel to take effect. 

```python
import samgeo
samgeo.update_package()
```
